Enable protection tests

jsafrane opened the pull request:
**What this PR does / why we need it**:

- StorageObjectInUseProtection feature is enabled by default so the test can run in regular e2e test suite
- Rename PVC protection test, it tests only PVCs and not whole storage.


**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

NickrenREN commented:
/lgtm

jsafrane commented:
/retest

jsafrane commented:
umm, the pv protection test did not pass... I'll look at it tomorrow.

jsafrane commented:
@NickrenREN, can you look at the failures when you get the chance?

NickrenREN commented:
yeah, will take a look

NickrenREN commented:
I tested it locally and specifically output the pv finalizer
```
[sig-storage] PV Protection 
  Verify that PV bound to a PVC is not removed immediately
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/pv_protection.go:104
[BeforeEach] [sig-storage] PV Protection
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:141
STEP: Creating a kubernetes client
Feb 23 10:04:46.211: INFO: >>> kubeConfig: /home/nickren/.kube/config
STEP: Building a namespace api object
STEP: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [sig-storage] PV Protection
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/pv_protection.go:50
Feb 23 10:04:46.242: INFO: Waiting up to 4h0m0s for all (but 0) nodes to be schedulable
STEP: Creating a PV
Feb 23 10:04:46.244: INFO: PV ReclaimPolicy unspecified, default: Retain
STEP: Checking that PV Protection finalizer is set
**pv finalizers is: [kubernetes.io/pv-protection]**
[It] Verify that PV bound to a PVC is not removed immediately
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/pv_protection.go:104
STEP: Creating a PVC
Feb 23 10:04:46.250: INFO: AccessModes unspecified, default: all modes (RWO, RWX, ROX).
STEP: Waiting for PVC to become Bound
Feb 23 10:04:46.254: INFO: Waiting up to 3m0s for PersistentVolumeClaim pvc-r5lw7 to have phase Bound
Feb 23 10:04:46.258: INFO: PersistentVolumeClaim pvc-r5lw7 found but phase is Pending instead of Bound.
Feb 23 10:04:48.259: INFO: PersistentVolumeClaim pvc-r5lw7 found and phase=Bound (2.004702239s)
STEP: Deleting the PV, however, the PV must not be removed from the system as it's bound to a PVC
STEP: Checking that the PV status is Terminating
STEP: Deleting the PVC that is bound to the PV
STEP: Checking that the PV is automatically removed from the system because it's no longer bound to a PVC
Feb 23 10:04:48.270: INFO: Waiting up to 3m0s for PersistentVolume hostpath-95xdm to get deleted
Feb 23 10:04:48.280: INFO: PersistentVolume hostpath-95xdm found and phase=Released (742.583µs)
Feb 23 10:04:50.282: INFO: PersistentVolume hostpath-95xdm was removed
[AfterEach] [sig-storage] PV Protection
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:142
Feb 23 10:04:50.282: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "e2e-tests-pv-protection-8khd6" for this suite.
Feb 23 10:04:56.792: INFO: Waiting up to 30s for server preferred namespaced resources to be successfully discovered
Feb 23 10:04:56.862: INFO: namespace: e2e-tests-pv-protection-8khd6, resource: bindings, ignored listing per whitelist
Feb 23 10:04:56.879: INFO: namespace e2e-tests-pv-protection-8khd6 deletion completed in 6.594035436s
[AfterEach] [sig-storage] PV Protection
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/pv_protection.go:90
Feb 23 10:04:56.879: INFO: AfterEach: Cleaning up test resources.
Feb 23 10:04:56.880: INFO: Deleting PersistentVolumeClaim "pvc-r5lw7"
Feb 23 10:04:56.881: INFO: Deleting PersistentVolume "hostpath-95xdm"

• [SLOW TEST:10.670 seconds]
[sig-storage] PV Protection
/home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/utils/framework.go:22
  Verify that PV bound to a PVC is not removed immediately
  /home/nickren/GoProjects/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/pv_protection.go:104
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSFeb 23 10:04:56.883: INFO: Running AfterSuite actions on all node
Feb 23 10:04:56.883: INFO: Running AfterSuite actions on node 1

Ran 2 of 729 Specs in 18.918 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 727 Skipped PASS

Ginkgo ran 1 suite in 19.286676142s
Test Suite Passed
2018/02/23 10:04:56 util.go:174: Step './hack/ginkgo-e2e.sh --ginkgo.focus=PV.Protection' finished in 19.336146047s
2018/02/23 10:04:56 e2e.go:81: Done
nickren@nickren-14:~/GoProjects/src/k8s.io/kubernetes$ 
```

NickrenREN commented:
debug: sleep 5s for finalizer to be added  https://github.com/kubernetes/kubernetes/pull/60264

NickrenREN commented:
yeah, if we sleep 5s for pv finalizer to be added, all checks are green. that may be the cause

msau42 commented:
What if you wait for the finalizer to be added before proceeding with the rest of the test

jsafrane commented:
Added a commit that enables StorageObjectInUseProtection admission plugin in tests.

NickrenREN commented:
/lgtm

jsafrane commented:
/retest

jsafrane commented:
/assign @wojtek-t @gmarek 
for approval

Note that StorageObjectInUseProtection admission plugin is already enabled in all other places I could find.

gmarek commented:
Please squash commits.

jsafrane committed:
Enable PV protection test by default

- StorageObjectInUseProtection is enabled by default now so the test can run in regular tests.
- Enable StorageObjectInUseProtection admission plugins during tests

jsafrane commented:
@gmarek squashed, PTAL.

msau42 commented:
/lgtm

NickrenREN commented:
/assign @gmarek for approval

wojtek-t commented:
/approve no-issue
